src/api/vade_evan_api.rs

```diff
@@ -486,6 +486,71 @@ impl VadeEvan {
             .map_err(|err| err.into())
     }
 
+    /// Creates a new zero-knowledge proof self issued credential.
+    /// `create_self_issued_credential` function can be used in the same step
+    /// and produces the same output as `vc_zkp_create_credential_offer` but uses a simpler argument setup.
+    ///
+    /// # Arguments
+    ///
+    /// * `schema_did` - schema to create the credential
+    /// * `issuer_did` - DID of issuer
+    /// * `subject_did` - DID of subject
+    ///
+    /// # Returns
+    /// * credential as JSON serialized [`BbsCredential`](https://docs.rs/vade_evan_bbs/*/vade_evan_bbs/struct.BbsCredential.html)
+    /// # Example
+    ///
+    /// ```
+    /// cfg_if::cfg_if! {
+    ///     if #[cfg(not(all(feature = "target-c-lib", feature = "capability-sdk")))] {
+    ///         use anyhow::Result;
+    ///         use vade_evan::{VadeEvan, VadeEvanConfig, DEFAULT_TARGET, DEFAULT_SIGNER};
+    ///
+    ///         const SCHEMA_DID: &str = "did:evan:EiACv4q04NPkNRXQzQHOEMa3r1p_uINgX75VYP2gaK5ADw";
+    ///         const CREDENTIAL_SUBJECT_STR: &str = "{\"id\":\"did:evan:testcore:0x67ce8b01b3b75a9ba4a1462139a1edaa0d2f539f\", \"data\":{\"test\":\"value\"}";
+    ///         const BBS_SECRET: &str = "bbssecret";
+    ///         const BBS_PRIVATE_KEY: &str = "bbsprivkey";
+    ///
+    ///         async fn example() -> Result<()> {
+    ///             let mut vade_evan = VadeEvan::new(VadeEvanConfig { target: DEFAULT_TARGET, signer: DEFAULT_SIGNER })?;
+    ///             let offer_str = vade_evan
+    ///                 .helper_create_self_issued_credential(
+    ///                     SCHEMA_DID,
+    ///                     CREDENTIAL_SUBJECT_STR,
+    ///                     BBS_SECRET,
+    ///                     BBS_PRIVATE_KEY,
+    ///                     Some(""),
+    ///                     Some(""),
+    ///                     Some(""),
```

Comment:
Is this valid input for an expected behaviour?

```rs
        credential_revocation_did: Option<&str>,
        credential_revocation_id: Option<&str>,
        exp_date: Option<&str>,
```

So `credential_revocation_did` and `credential_revocation_id` seem to be used, as they are provided - empty strings, which does not look like valid values. :D

And `exp_date` as `""` should turn on `use_valid_until` and use `""` as the value for `valid_until`. Looks like this would produce a credential with values that do not make too much sense. :D

Reply:
Fixed, thanks, I did it at the very beginning of my 'rusting' :)  Also fixed the function description.  Will submit all together.